src/overlays/actors/ovl_Bg_Ydan_Hasi/z_bg_ydan_hasi.c

```diff
@@ -9,7 +14,13 @@ void BgYdanHasi_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void BgYdanHasi_Update(Actor* thisx, GlobalContext* globalCtx);
 void BgYdanHasi_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
+void BgYdanHasi_InitWater(BgYdanHasi* this, GlobalContext* globalCtx);
+void BgYdanHasi_UpdateFloatingBlock(BgYdanHasi* this, GlobalContext* globalCtx);
+void BgYdanHasi_SetupThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx);
+WaterBox* BgYdanHasi_MoveWater(BgYdanHasi* this, GlobalContext* globalCtx);
+void BgYdanHasi_DecWaterTimer(BgYdanHasi* this, GlobalContext* globalCtx);
+void BgYdanHasi_UpdateThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx);
```

Comment:
in general these names arent really named after actions. in this case its definitley hard to name it after some sort of action, but I would try to if you can.
For example BgYdanHasi_UpdateFloatingBlock -> BgYdanHasi_FloatPlatform ? 
The action is floating the platform on the water